### ISSUE TITLE
unexpected not match regex result

### ISSUE THREAD

**Issue by mxlxm** [VictoriaMetrics/VictoriaMetrics#546]
**Describe the bug**
not match regex returned unexpected result.
**To Reproduce**
container_memory_usage_bytes{container_name!~"|POD"}
got unexpected metrics with label `container_name=""`
**Expected behavior**
only return metrics with label `container_name != ""` AND `container_name != "POD"`
**Version**
master

**Comment by mxlxm** [comment]
tested, but the bug still exists.

query: `container_memory_usage_bytes{container_name!~"|POD"}`
still got unexpected result with label `container_name=""`,while, actually those metrics doesn't have this label.

```
./vmstorage -version
vmstorage-20200611-070747-heads-cluster-0-g35191d84
```

**Comment by mxlxm** [comment]
BTW, here's another bad new. 
After upgrade the whole cluster from v1.34.9 to `head cluster` with default `replicationFactor `, all data been rerouted to a certain vmstorage, then some query may cause vmstorage OOM, after OOM reroute to next vmstorage, repeatedly.
<img width="1627" alt="image" src="https://user-images.githubusercontent.com/1845326/84558796-ee54e800-ad67-11ea-9f0e-2a9f01dfb4f0.png">
 Rollback to v1.34.9, the cluster works as expected.

**Comment by valyala** [comment]
> tested, but the bug still exists

The bug must be fixed in the commit 5f3a895c239cee1a0359866e4da57fe69cfa4ad4 .

> After upgrade the whole cluster from v1.34.9 to head cluster with default replicationFactor , all data been rerouted to a certain vmstorage,

Confirmed the bug. It must be fixed in the commit 464682f380501b87555f7f280ba5006d4d60a373 .

**Comment by valyala** [comment]
FYI, all the bugfixes mentioned above are included in VictoriaMetrics [v1.37.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Comment by mxlxm** [comment]
I'll verify this on weekend, a little busy today.
Thanks again.
